### PR TITLE
Remove regex

### PR DESCRIPTION
**Pull request by btipling**

### CODE REVIEW

File: pgdb/v1/fts.go [hunk 1]
```diff
@@ -136,14 +115,47 @@ func camelSplitDocs(docs []*SearchContent) []lexeme {
...
+		var prev rune
+		for i, r := range docValue {
+			if prev == 0 {
+				prev = r
+				continue
+			}
+			if unicode.IsUpper(prev) {
+				if unicode.IsSpace(r) && len(word) > 0 {
+					word = append(word, prev)
+					rv = append(rv, lexeme{strings.ToLower(string(word)), i - len(word), doc.Weight})
```
Comment (btipling) [inline pgdb/v1/fts.go:143]: no +1 as we're always 1 ahead

File: pgdb/v1/fts.go [hunk 2]
```diff
@@ -136,14 +115,47 @@ func camelSplitDocs(docs []*SearchContent) []lexeme {
 			continue
 		}
 		docValue := interfaceToValue(doc.Value)
-		for _, match := range camelCaseRegex.FindAllStringIndex(docValue, -1) {
-			result := docValue[match[0]:match[1]]
-			rv = append(rv, lexeme{strings.ToLower(result), match[0] + 1, doc.Weight})
+		var word []rune
+		for i, r := range docValue {
```
Comment (pquerna) [inline pgdb/v1/fts.go:119]: can we make this a small function that we can directly unit test?

Comment (btipling) [reply pgdb/v1/fts.go]: @pquerna tests and fixes for things tests found: in 93e4ee3ab1cca0577c7e72b3efa1ebd4c01b81f6

Comment (btipling) [reply pgdb/v1/fts.go]: Added even more tests

File: pgdb/v1/fts.go [hunk 3]
```diff
@@ -13,32 +12,14 @@ import (
...
-
-const template = "$token"
-
 type SearchContent struct {
 	Type   FieldOptions_FullTextType
 	Weight FieldOptions_FullTextWeight
 	Value  interface{}
 }
 
+const MinWordSize = 3
```
Comment (pquerna) [inline pgdb/v1/fts.go:21]: shouldn't be exported

File: pgdb/v1/fts.go [hunk 4]
```diff
@@ -129,22 +110,99 @@ func lemmatizeDocs(docs []*SearchContent, additionalFilters ...jargon.Filter) []
 	return rv
 }
 
+func camelSplitDoc(docValue string, doc *SearchContent) []lexeme {
+	rv := make([]lexeme, 0, 8)
+	var word []rune
```
Comment (pquerna) [inline pgdb/v1/fts.go:115]: perf wise, wonder if this should be using a `bytes.Buffer` with Reset() rather than `nil` the slice

Comment (btipling) [reply pgdb/v1/fts.go]: @pquerna changed in 964f2e6fbd331553c6bd49d3c749e968269e88bd

File: pgdb/v1/fts.go (hunk 3 above)
Comment (btipling) [reply pgdb/v1/fts.go]: changed in 87da3bca39151e296b78e05bdbc51cd0e7d6099e

File: pgdb/v1/fts.go [hunk 5]
```diff
@@ -129,22 +112,117 @@ func lemmatizeDocs(docs []*SearchContent, additionalFilters ...jargon.Filter) []
...
+		if prev == 0 {
+			prev = r
+			continue
+		}
+		if unicode.IsUpper(prev) {
+			if buffer.Len() == 0 { // no current word
+				if unicode.IsLower(r) {
+					// got a upper case in prev and current is lower, starting a new word
+					if _, e := buffer.WriteRune(prev); e != nil {
+						buffer = bytes.Buffer{}
```
Comment (pquerna) [inline pgdb/v1/fts.go:130]: shouldn't this be buffer.Reset() so the underlying slice is reused?

Comment (btipling) [reply pgdb/v1/fts.go]: Oh right ec2a9f7d9a9bd0e6b7225ab81d52fd7784bd276c

**Review by pquerna — APPROVED** [review]

**Pull request merged**